src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,18 +3,40 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE_SIZE = 10;
+    private final Object[] keyStorage = new Object[MAX_STORAGE_SIZE];
+    private final Object[] valueStorage = new Object[MAX_STORAGE_SIZE];
+    private int count = 0;
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < count; i++) {
+            if (key == null && keyStorage[i] == null || key != null && key.equals(keyStorage[i])) {
```

Comment:
```suggestion
            if (key == keyStorage[i] || key != null && key.equals(keyStorage[i])) {
```
you can shorten this condition, because if they are both null '==' comparison is valid. 
BTW:
the best way to write this condition is to use Objects.equals(key, keyStorage[i]). We just forbid it in checklist so you can learn how to write this check manually properly. But if you understand already how to write it with null check you use Obejcts.equals